presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -92,6 +92,62 @@ Additionally, the relations within a ``WITH`` clause can chain::
     relation is used. This means that if the relation is used more than once and the query
     is non-deterministic, the results may be different each time.
 
+
+WITH RECURSIVE Clause
+---------------------
+
+The ``WITH RECURSIVE`` clause is a variant of the ``WITH`` clause. It defines a
+``UNION`` of two relations. The first relation is called the *recursion base*,
+and the second relation is called the *recursion step*:
```

Comment:
It is not correct to say that `WITH RECURSIVE` clause defines a union of relations.

After the words `WITH RECURSIVE`, there comes a _list_ of queries. Each of those queries is _potentially recursive_. Some might be actually recursive, some might be not. Each of those queries which are actually recursive, defines a union of two relations.

We know that a query in the list is actually recursive when it meets certain conditions, roughly speaking:
- it is shaped as union,
- it has a recursive reference to its own name.

Reply:
Updated .. please verify